### DISCUSSION TITLE
gitbook notes

### DISCUSSION THREAD

**Discussion by devrdn** [GitbookIO/community#492] [Feature Requests]
Hey folks,

Got a question.

In git, can you drop notes like this

```
> [!NOTES]
> ...
```
Is it possible to adapt this markdown for GitBook?

**Comment by addisonschultz** [comment]
Hey @devrdn - great idea! I'm curious, would using comments help for what you're trying to do potentially?

**Reply by devrdn** [reply]
Suppose there is project documentation where certain important points are highlighted in markdown using "notes." In GitHub, they can be created like this:

```
> [!NOTE]
> ...

> [!IMPORTANT]
> ...
```

GitBook does not interpret them as its own notes.

It would be desirable for GitBook to understand them.

Github:
![image](https://github.com/GitbookIO/community/assets/66188093/fb4daba5-75db-42e0-873a-118019bd3854)

GitBook:
![image](https://github.com/GitbookIO/community/assets/66188093/64fc0bb7-2214-4d02-a497-17e68f9cb81b)

We want (smthng like this):
![image](https://github.com/GitbookIO/community/assets/66188093/e12eb928-1874-4262-9cec-ffef1beacde5)

**Reply by addisonschultz** [reply]
I see what you mean. We have hint blocks in GitBook, which would solve for this - but unfortunately we use a different syntax to render them. 

Check this out for more info: https://docs.gitbook.com/content-editor/blocks/hint#representation-in-markdown

**Reply by devrdn** [reply]
Yep, right, i know!

But if, for example, I want the documentation to look good both here and there? Are there any integrations or plugins that allow for synchronizing this? Often this is very important

**Reply by devrdn** [reply]
And, can I set up additional plugins in new versions of GitBook?

**Reply by addisonschultz** [reply]
We do have an [integration platform](https://developer.gitbook.com/) that allows you to build apps and workflows into the way you're already working. You should be able to combine some APIs with [GitSync](https://docs.gitbook.com/integrations/git-sync) to transform the syntax for you. 

That being said, we're also working on more ways for you to customize the way you render content in GitHub, which also might help. More on that coming from us soon!

**Reply by devrdn** [reply]
So, is it possible to configure it in such a way that this syntax is correctly recognized with the help of Git Sync integration? At the moment, this aspect is very important for out project

**Reply by addisonschultz** [reply]
It should be possible but would need to be built custom - using the integration platform I linked above! Unfortunately right now we don't have any out of the box solutions that would solve for this

**Reply by devrdn** [reply]
So, this is a plugin available, but idk if this is possible to install it, with new version of gitbook 🤔

**Reply by devrdn** [reply]
https://github.com/fzankl/gitbook-plugin-flexible-alerts

**Reply by addisonschultz** [reply]
Hi @devrdn! That plugin is unfortunately for an older (deprecated) version of GitBook - so that plugin won't work for this scenario

**Reply by devrdn** [reply]
thanks, for help 
i'll try to write an integration if it possible

**Reply by devrdn** [reply]
> We do have an [integration platform](https://developer.gitbook.com/) that allows you to build apps and workflows into the way you're already working. You should be able to combine some APIs with [GitSync](https://docs.gitbook.com/integrations/git-sync) to transform the syntax for you.
> 
> That being said, we're also working on more ways for you to customize the way you render content in GitHub, which also might help. More on that coming from us soon!

hi, folk!

so, can you provide some links, where i can find some info about creating this integration: transform the syntax for you?

and, can you tell me if this is possible in general or not?

**Reply by SamyPesse** [reply]
It's not possible to create integrations to extend this type of syntax. But this is something we should support in our core markdown parser.
We currently use https://github.com/remarkjs/remark, but it doesn't look like it supports this syntax yet. We'll see if we can prioritize work to support it.

**Reply by devrdn** [reply]
> It's not possible to create integrations to extend this type of syntax. But this is something we should support in our core markdown parser. We currently use https://github.com/remarkjs/remark, but it doesn't look like it supports this syntax yet. We'll see if we can prioritize work to support it.

It would be really cool because sometimes you have to store documentation both in Git and GitBook, and alerts are a very useful feature for highlighting text, so I wouldn’t want to deal with two different markup formats.

Thanks! :)

**Reply by devrdn** [reply]
> It's not possible to create integrations to extend this type of syntax. But this is something we should support in our core markdown parser. We currently use https://github.com/remarkjs/remark, but it doesn't look like it supports this syntax yet. We'll see if we can prioritize work to support it.

Sorry for the inconvenience, but if any work in this direction is initiated, could you inform me about it? It’s very important to me.

If you need any additional information or some help, please don't hesitate to write.


So, for example, there are some plugins for remark.
- https://github.com/hyoban/remark-github-alerts
- https://github.com/chrisweb/rehype-github-alerts
- https://github.com/jaywcjlove/remark-github-blockquote-alert
- https://github.com/myl7/remark-github-beta-blockquote-admonitions

Some of these are added to the official plugins list: https://github.com/remarkjs/remark/blob/main/doc/plugins.md#list-of-plugins 

Thank you so much once again 😎

**Comment by devrdn** [comment]
Hey, any updates on this?

**Reply by addisonschultz** [reply]
Hi @devrdn - unfortunately there hasn’t been any movement on this feature request, but I’ve moved this to our feature requests section where we can see if this feature would be useful to more of the GitBook community! If we do end up working on this, we’ll update you here.
